cypress/support/step_definitions/validations.js

```diff
@@ -64,6 +64,17 @@ Given(
           const expectingError = item.expectingError;
           const isNullCase = item.isNullCase || false;
 
+          for (let i = 0; i < contentObject.data.length; i++) {
+            if (contentObject.data[i].validations[0].mode == CONSTANTS.DEVICE_CONTENT_VALIDATION) {
+              const type = contentObject.data[i].validations[0].type;
+              if (Cypress.env(CONSTANTS.DEVICE_DATA).hasOwnProperty(type)) {
+                contentObject.data[i].validations[0].type = Cypress.env(CONSTANTS.DEVICE_DATA)[
+                  type
+                ];
+              }
+            }
+          }
+
           // If the app ID is not passed from the feature, the default app ID will be retrieved.
```

Comment:
This would not work for localization. Only few Api's are having validation,type == 'deviceContentValidation'. 

Ideally, we should define the structure in this manner. 
if (source of truth fetch status == true && (CONSTANTS.DYNAMIC_DEVICE_DETAILS_MODULES. includes(Cypress.env(CONSTANTS.TEST_TYPE))) {) {
    search for env with the name as we use for validation and use that for validation 
}

Reply:
removed deviceContentValidation check